wwwroot/inc/upgrade.php

```diff
@@ -1312,6 +1312,7 @@ function getUpgradeBatch ($batchid)
 			$query[] = "ALTER TABLE MountOperation DROP KEY `MountOperation-FK-new_molecule_id`";
 			$query[] = "ALTER TABLE IPv4Allocation MODIFY TYPE ENUM('regular','shared','virtual','router','point2point', 'sharedrouter') NOT NULL DEFAULT 'regular';";
 			$query[] = "ALTER TABLE IPv6Allocation MODIFY TYPE ENUM('regular','shared','virtual','router','point2point', 'sharedrouter') NOT NULL DEFAULT 'regular';";
+			$query[] = "INSERT INTO AttributeMap (`objtype_id`,`attr_id`,`chapter_id`) VALUES (12,2,39)"; // UPS (UPS models) -> HW type
```

Comment:
This query will fail unless `INSERT INTO Chapter ...` is done first. Would you like to fix it yourself?

Reply:
Added the insert query